### PR TITLE
State race backport

### PR DESCRIPTION
**Pull request by vyzaldysanchez**

### CODE REVIEW

**Comment by vyzaldysanchez** [comment]
Raising this PR to backport the state race fix from @jmank88

**Review by samsondav — APPROVED** [review]

File: core/chains/evm/client/node.go [hunk 1]
```diff
@@ -288,10 +288,11 @@ func (n *node) verify(callerCtx context.Context) (err error) {
 		promEVMPoolRPCNodeVerifiesFailed.WithLabelValues(n.chainID.String(), n.name).Inc()
 	}
 
-	switch n.state {
+	st := n.State()
```
Comment (samsondav) [inline core/chains/evm/client/node.go:291]: wait, this should always be done inside a mutex. Is there a case where it is not? See comment on line 280

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/chains/evm/client/node.go (hunk 1 above)
Comment (jmank88) [reply core/chains/evm/client/node.go:291]: None of the callers are locking the mutex. In the one case of `start` that was fine, but he others were racey and one was flagged by the detector in a local run which led me here.
Most of the callers make other calls to methods which lock themseles, including the corresponding `State()` method which does it's own locking, so this seemed to fit local style but I overlooked the comment. Feel free to fix either way, but FWIW locking as tightly as possible is usually favorable.

**Review by samsondav — APPROVED** [review]

File: core/chains/evm/client/node.go (hunk 1 above)
Comment (samsondav) [reply core/chains/evm/client/node.go:291]: Yup, looks reasonable

**Pull request merged**